### ISSUE TITLE
Add way for comparing EPrints versions to see which is more recent

### ISSUE THREAD

**Issue by drn05r** [eprints/eprints3.4#90]
With Accessibility changes it will become necessary for ingredients and plugins to be able to behave slighly differenty depending on whether they are working with a pre or post Accessibility changes version.  Currently there is no way of checking whether this without a rather bespoke and complex conditional statement.  Ideally we should have something that can do this more concisely and hides the more complex logic.

Current a version is styled as follows: vX.Y.Z but can also contain a hyphen followed by rc1, rc2 etc.  It may also be used to display alpha1 or beta2, etc.  This being the case if there is a hyphen this is considered to be before the non-hyphened version.  As alpha would come before beta and that before rc then we can just use an alphanumeric check for after the hyphen.  Therefore, altjough the check might be complex, it should be able to always determine whether any particular version is greater or less than the current version.
